core/client/fs/src/main/java/alluxio/client/file/AlluxioFileInStream.java

```diff
@@ -380,59 +376,64 @@ private void closeBlockInStream(BlockInStream stream) throws IOException {
       }
-      triggerAsyncCaching(stream);
+      // best effort to cache this block
+      try {
+        triggerAsyncCaching(stream);
+      } catch (Exception e) {
+        LOG.warn("Failed to complete async cache request (best effort) for block {} of file {}: {}",
+            stream.getId(), mStatus.getPath(), e.toString());
+      }
     }
   }
 
   // Send an async cache request to a worker based on read type and passive cache options.
-  // Note that, this is best effort
-  private void triggerAsyncCaching(BlockInStream stream) {
-    try {
-      boolean cache = ReadType.fromProto(mOptions.getOptions().getReadType()).isCache();
-      boolean overReplicated = mStatus.getReplicationMax() > 0
-          && mStatus.getFileBlockInfos().get((int) (getPos() / mBlockSize))
-          .getBlockInfo().getLocations().size() >= mStatus.getReplicationMax();
-      cache = cache && !overReplicated;
-      // Get relevant information from the stream.
-      WorkerNetAddress dataSource = stream.getAddress();
-      long blockId = stream.getId();
-      if (cache && (mLastBlockIdCached != blockId)) {
-        // Construct the async cache request
-        long blockLength = mOptions.getBlockInfo(blockId).getLength();
-        String host = dataSource.getHost();
-        // issues#11172: If the worker is in a container, use the container hostname
-        // to establish the connection.
-        if (!dataSource.getContainerHost().equals("")) {
-          LOG.debug("Worker is in a container. Use container host {} instead of physical host {}",
-              dataSource.getContainerHost(), host);
-          host = dataSource.getContainerHost();
-        }
-        AsyncCacheRequest request =
-            AsyncCacheRequest.newBuilder().setBlockId(blockId).setLength(blockLength)
-                .setOpenUfsBlockOptions(mOptions.getOpenUfsBlockOptions(blockId))
-                .setSourceHost(host).setSourcePort(dataSource.getDataPort())
-                .build();
-        if (mPassiveCachingEnabled && mContext.hasProcessLocalWorker()) {
-          mContext.getProcessLocalWorker().asyncCache(request);
-          mLastBlockIdCached = blockId;
-          return;
-        }
-        WorkerNetAddress worker;
-        if (mPassiveCachingEnabled && mContext.hasNodeLocalWorker()) {
-          // send request to local worker
-          worker = mContext.getNodeLocalWorker();
-        } else { // send request to data source
-          worker = dataSource;
-        }
-        try (CloseableResource<BlockWorkerClient> blockWorker =
-                 mContext.acquireBlockWorkerClient(worker)) {
-          blockWorker.get().asyncCache(request);
-          mLastBlockIdCached = blockId;
-        }
+  // Note that, this is best effort and exception shall be caught and ignored
+  @VisibleForTesting
+  void triggerAsyncCaching(BlockInStream stream) throws Exception {
+    final long blockId = stream.getId();
+    final BlockInfo blockInfo = mStatus.getBlockInfo(blockId);
```

Comment:
this is the only meaningful change in this PR, compared to previous `mStatus.getFileBlockInfos().get((int) (getPos() / mBlockSize))`